opentracing/tracecontext.go

```diff
@@ -63,10 +90,31 @@ type TraceContextMarshaler interface {
 type TraceContextUnmarshaler interface {
 	// Converts the marshaled binary data (see
 	// TraceContextMarshaler.MarshalTraceContextBinary()) into a TraceContext.
-	UnmarshalTraceContextBinary(marshaled []byte) (TraceContext, error)
+	//
+	// The first parameter contains the marshaler's serialization of the core
+	// identifying information in a TraceContext instance.
+	//
+	// The first parameter contains the marshaler's serialization of the trace
+	// tags (per `SetTraceTag` and `TraceTag`) attached to a TraceContext
+	// instance.
+	UnmarshalTraceContextBinary(
+		traceContextID []byte,
+		traceTags []byte,
+	) (TraceContext, error)
+
 	// Converts the marshaled string:string map (see
 	// TraceContextMarshaler.MarshalTraceContextStringMap()) into a TraceContext.
-	UnmarshalTraceContextStringMap(marshaled map[string]string) (TraceContext, error)
+	//
+	// The first parameter contains the marshaler's serialization of the core
+	// identifying information in a TraceContext instance.
+	//
+	// The first parameter contains the marshaler's serialization of the trace
+	// tags (per `SetTraceTag` and `TraceTag`) attached to a TraceContext
+	// instance.
+	UnmarshalTraceContextStringMap(
```

Comment:
Perhaps add a comment that it's ok to pass the same map as a value for both parameters, e.g. http headers map. The method only takes those keys from the map that it deems relevant.